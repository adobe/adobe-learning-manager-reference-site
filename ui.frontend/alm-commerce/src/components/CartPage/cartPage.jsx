/**
Copyright 2021 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

import React from "react";
import { useCartPage } from "../../hooks/CartPage/useCartPage";
import CommerceLoader from "../Common/Loader";
import styles from "./cartPage.module.css";
import OrderSummary from "./orderSummary";
import ProductList from "./productList";

const CartPage = (props) => {
  const {
    cartItems,
    hasItems,
    shouldShowLoadingIndicator,
    prices = {},
    proceedToCheckout,
    refreshCartHandler,
  } = useCartPage();

  const totalPrice = prices["grand_total"] || {};

  if (shouldShowLoadingIndicator) {
    return <CommerceLoader size="L"></CommerceLoader>;
  }
  return (
    <div className={styles.pageContainer}>
      <h1>Shopping Cart</h1>
      <div className={styles.cartPageContainer}>
        <div className={styles.productList}>
          <ProductList
            cartItems={cartItems}
            refreshCartHandler={refreshCartHandler}
          />
        </div>
        {hasItems && (
          <div className={styles.orderSummary}>
            <OrderSummary
              buttonLabel={"Proceed to Checkout"}
              clickHandler={proceedToCheckout}
              totalPrice={totalPrice}
            />
          </div>
        )}
      </div>
    </div>
  );
};

export default CartPage;

/**
 * 
 // console.log(GET_CART_DETAILS);

  const talonProps = useCartPage();

  //   if (loading) return "Loading...";
  //   if (error) return `Error! ${error.message}`;
  //   const talonProps = useCartPage({
  //     operations: getCartDetailsQuery,
  //   });

  const {
    cartItems,
    hasItems,
    isCartUpdating,
    fetchCartDetails,
    onAddToWishlistSuccess,
    setIsCartUpdating,
    shouldShowLoadingIndicator,
    wishlistSuccessProps,
  } = talonProps;

  const classes = useStyle(defaultClasses, props.classes);
  const [, { addToast }] = useToasts();

  useEffect(() => {
    if (wishlistSuccessProps) {
      addToast({ ...wishlistSuccessProps, icon: CheckIcon });
    }
  }, [addToast, wishlistSuccessProps]);

  if (shouldShowLoadingIndicator) {
    return fullPageLoadingIndicator;
  }

  const productListing = hasItems ? (
    <ProductListing
      onAddToWishlistSuccess={onAddToWishlistSuccess}
      setIsCartUpdating={setIsCartUpdating}
      fetchCartDetails={fetchCartDetails}
    />
  ) : (
    <h3>
      <FormattedMessage
        id={"cartPage.emptyCart"}
        defaultMessage={"There are no items in your cart."}
      />
    </h3>
  );

  const priceAdjustments = hasItems ? (
    <PriceAdjustments setIsCartUpdating={setIsCartUpdating} />
  ) : null;

  const priceSummary = hasItems ? (
    <PriceSummary isUpdating={isCartUpdating} />
  ) : null;

  return (
    <div className={classes.root}>
      <div className={classes.heading_container}>
        <h1 className={classes.heading}>
          <FormattedMessage id={"cartPage.heading"} defaultMessage={"Cart"} />
        </h1>
        <div className={classes.stockStatusMessageContainer}>
          <StockStatusMessage cartItems={cartItems} />
        </div>
      </div>
      <div className={classes.body}>
        <div className={classes.items_container}>{productListing}</div>
        <div className={classes.price_adjustments_container}>
          {priceAdjustments}
        </div>
        <div className={classes.summary_container}>
          <div className={classes.summary_contents}>{priceSummary}</div>
        </div>
      </div>
    </div>
  );
 */
