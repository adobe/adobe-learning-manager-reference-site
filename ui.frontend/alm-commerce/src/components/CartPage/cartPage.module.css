/*
Copyright 2021 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

@import "../../styles/variables.css";

.pageContainer {
  /* max-width: var(--prime-container-max-width-desktop);
  width: var(--prime-container-width);
  margin: 0 auto; */
}

.productList {
  width: 70%;
  padding-right: 30px;
  padding-bottom: 30px;
}
.cartPageContainer {
  display: flex;
}
h1 {
  font-size: var(--prime-font-size-28);
  color: var(--prime-color-black);
}

.orderSummary {
  width: 30%;
  /* padding: 20px 0; */
}

@media screen and (max-width: 450px) {
  .productList {
    /* order: 2; */
    width: 100%;
    padding: 20px;
  }
  .orderSummary {
    /* order: 1; */
    width: 100%;
    padding: 20px 20px 0 20px;
  }
  .pageContainer {
    width: 100%;
    max-width: var(--prime-container-max-width-mobile);
  }
  .cartPageContainer {
    flex-direction: column;
  }
}
