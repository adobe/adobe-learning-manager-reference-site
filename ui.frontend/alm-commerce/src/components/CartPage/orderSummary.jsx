/**
Copyright 2021 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/
import React from 'react'
import styles from "./orderSummary.module.css";
import { formatPrice } from "../../utils/price";

export default function OrderSummary(props) {

    const { buttonLabel, clickHandler, totalPrice } = props
    return (
        <div className={styles.summaryContainer}>
            <h2>Summary</h2>
            <hr />
            <div className={styles.orderTotal}>
                <span>Order Total</span>
                <span>{formatPrice(totalPrice)}</span>
            </div>

            <div className={styles.actionContainer}>
                <button variant="cta" type="button" onClick={clickHandler} className={`almButton primary`}>
                    {buttonLabel}
                </button>
            </div>
        </div >
    )
}
