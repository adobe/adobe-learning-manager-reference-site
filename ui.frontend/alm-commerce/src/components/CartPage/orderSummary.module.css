/**
Copyright 2021 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/
@import "../../styles/variables.css";

.summaryContainer {
  background-color: var(--prime-color-nc6);
  padding: 20px;
  font-size: var(--prime-font-size-21);
}

h2 {
  font-size: var(--prime-font-size-21);
  margin: 0;
}
.orderTotal {
  margin: 20px 0;
  display: flex;
  justify-content: space-between;
  font-size: var(--prime-font-size-18);
  color: var(--prime-color-black);
  font-weight: 600;
}

.actionContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 40px;
}
.actionContainer button{
  width: 100%;
  font-size: var(--prime-font-size-16);
}