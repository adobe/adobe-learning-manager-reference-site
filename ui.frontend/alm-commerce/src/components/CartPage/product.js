import React from 'react'

export default function Product(props) {
  const { name, almthumbnailurl, price_range } = props.training;
  let finalPrice = price_range?.maximum_price?.final_price;
  let priceLabel = "";
  if (finalPrice) {
    const formatter = new Intl.NumberFormat("en-US", {
      style: "currency",
      currency: finalPrice.currency,
    });

    priceLabel = formatter.format(finalPrice.value);
  }
  return (
    <div>
      <img src={almthumbnailurl} alt="" />
      <div>{name}</div>
      <div>{priceLabel}</div>
    </div>
  )
}
