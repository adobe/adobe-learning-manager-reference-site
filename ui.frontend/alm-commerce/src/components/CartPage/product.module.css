/**
Copyright 2022 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

@import "../../styles/variables.css";

.itemContainer {
  display: flex;
  padding: 10px 0;
  gap: 20px;
}

.itemImage {
  width: 100px;
  height: 100px;
  background-color: grey;
  flex-direction: row;
  flex: 100px 0 0;
}
.detailsContainer {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  flex: 4 0 0;
}

.name {
  padding-right: 20px;
  font-size: var(--prime-font-size-16);
  color: var(--prime-color-black);
}

.removeItem {
  display: flex;
  justify-content: flex-end;
  float: right;
}

.removeItem svg {
  fill: var(--prime-color-nc3);
}

.priceLabel {
  display: flex;
  justify-content: flex-end;
  font-size: var(--prime-font-size-16);
  color: var(--prime-color-black);
}
