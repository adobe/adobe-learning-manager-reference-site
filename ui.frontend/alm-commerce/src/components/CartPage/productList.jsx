import React from 'react';
import Product from "./product";

export default function ProductList({ cartItems = [] }) {

    const productList = cartItems.map((training) => {
        return (
            <Product training={training.product} key={training.id} />
        )
    })
    return (
        cartItems.length ? (<div> {productList}</div>) : "No items in cart"
    )
}
