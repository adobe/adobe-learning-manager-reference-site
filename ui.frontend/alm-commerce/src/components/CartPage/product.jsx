import { Button } from "@adobe/react-spectrum";
import Delete from "@spectrum-icons/workflow/Delete";
import React from "react";
import { useProduct } from "../../hooks/CartPage/useProduct";
import { formatPrice } from "../../utils/price";
import { useCardBackgroundStyle } from "../../hooks/common/product";
import CommerceLoader from "../Common/Loader";
import styles from "./product.module.css";

export default function Product(props) {
  const { name, price_range } = props.training;
  let finalPrice = price_range?.maximum_price?.final_price;
  let priceLabel = formatPrice(finalPrice);
  const refreshCartHandler = props.refreshCartHandler;
  const { removeItemFromCart, loading } = useProduct(refreshCartHandler);
  const cardBackgroundStyle = useCardBackgroundStyle(props.training);
  const removeItemHandler = () => {
    removeItemFromCart(Number(props.itemId));
  };

  return (
    <div className={styles.itemContainer}>
      <img style={cardBackgroundStyle} alt="" className={styles.itemImage} />
      <div className={styles.detailsContainer}>
        <div className={styles.name}>{name}</div>
        <div>
          <div className={styles.priceLabel}>{priceLabel}</div>
          {props.canDeleteProduct &&
            <Button
              variant="overBackground"
              type="button"
              onPress={removeItemHandler}
              UNSAFE_className={styles.removeItem}
            >
              {loading ? <CommerceLoader size="S" /> : <Delete> </Delete>}
            </Button>
          }
        </div>
      </div>
    </div>
  );
}
