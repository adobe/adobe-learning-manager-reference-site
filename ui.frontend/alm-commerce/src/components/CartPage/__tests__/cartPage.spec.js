/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import ReactDOM from 'react-dom';
import { useCartPage } from '@magento/peregrine/lib/talons/CartPage/useCartPage';

import render from '../../utils/test-utils';
import CartPage from '../cartPage';

jest.mock('@magento/venia-ui/lib/components/StockStatusMessage', () => 'StockStatusMessage');
jest.mock('@magento/venia-ui/lib/components/CartPage/PriceAdjustments', () => 'PriceAdjustments');
jest.mock('../PriceSummary', () => 'PriceSummary');
jest.mock('../ProductListing', () => 'ProductListing');

jest.mock('@magento/peregrine/lib/talons/CartPage/useCartPage', () => {
    const useCartPageTalon = jest.requireActual('@magento/peregrine/lib/talons/CartPage/useCartPage');
    const spy = jest.spyOn(useCartPageTalon, 'useCartPage');

    return Object.assign(useCartPageTalon, { useCartPage: spy });
});

jest.mock('@magento/peregrine/lib/Toasts/useToasts', () => ({
    useToasts: () => [{}, { addToast: jest.fn() }]
}));

const talonProps = {
    hasItems: false,
    handleSignIn: jest.fn().mockName('handleSignIn'),
    isSignedIn: false,
    isCartUpdating: false,
    setIsCartUpdating: jest.fn().mockName('setIsCartUpdating'),
    shouldShowLoadingIndicator: false,
    onAddToWishlistSuccess: jest.fn().mockName('onAddToWishlistSuccess'),
    wishlistSuccessProps: null
};

describe('cartPage', () => {
    beforeAll(() => {
        /**
         * Mocking ReactDOM.createPortal because of incompatabilities
         * between ReactDOM and react-test-renderer.
         *
         * More info: https://github.com/facebook/react/issues/11565
         */
        ReactDOM.createPortal = jest.fn(element => {
            return element;
        });
    });

    afterAll(() => {
        ReactDOM.createPortal.mockClear();
    });

    test('renders a loading indicator when talon indicates', () => {
        // Arrange.
        const myTalonProps = {
            ...talonProps,
            shouldShowLoadingIndicator: true
        };
        useCartPage.mockReturnValueOnce(myTalonProps);

        // Act.
        const instance = render(<CartPage />);

        // Assert.
        expect(instance.toJSON()).toMatchSnapshot();
    });

    test('renders empty cart text (no adjustments, list or summary) if cart is empty', () => {
        // Arrange.
        useCartPage.mockReturnValueOnce(talonProps);

        // Act.
        const instance = render(<CartPage />);

        // Assert.
        expect(instance.toJSON()).toMatchSnapshot();
    });

    test('renders components if cart has items', () => {
        // Arrange.
        const myTalonProps = {
            ...talonProps,
            hasItems: true
        };
        useCartPage.mockReturnValueOnce(myTalonProps);

        // Act.
        const instance = render(<CartPage />);

        // Assert.
        expect(instance.toJSON()).toMatchSnapshot();
    });
});
