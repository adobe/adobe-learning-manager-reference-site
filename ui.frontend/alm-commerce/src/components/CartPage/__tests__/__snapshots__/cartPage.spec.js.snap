// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cartPage renders a loading indicator when talon indicates 1`] = `
<div>
  <span>
    <svg
      fill="none"
      height={64}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={64}
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="12"
        x2="12"
        y1="2"
        y2="6"
      />
      <line
        x1="12"
        x2="12"
        y1="18"
        y2="22"
      />
      <line
        x1="4.93"
        x2="7.76"
        y1="4.93"
        y2="7.76"
      />
      <line
        x1="16.24"
        x2="19.07"
        y1="16.24"
        y2="19.07"
      />
      <line
        x1="2"
        x2="6"
        y1="12"
        y2="12"
      />
      <line
        x1="18"
        x2="22"
        y1="12"
        y2="12"
      />
      <line
        x1="4.93"
        x2="7.76"
        y1="19.07"
        y2="16.24"
      />
      <line
        x1="16.24"
        x2="19.07"
        y1="7.76"
        y2="4.93"
      />
    </svg>
  </span>
  <span>
    Fetching Data...
  </span>
</div>
`;

exports[`cartPage renders components if cart has items 1`] = `
<div>
  <div>
    <h1>
      Cart
    </h1>
    <div>
      <StockStatusMessage />
    </div>
  </div>
  <div>
    <div>
      <ProductListing
        onAddToWishlistSuccess={[MockFunction onAddToWishlistSuccess]}
        setIsCartUpdating={[MockFunction setIsCartUpdating]}
      />
    </div>
    <div>
      <PriceAdjustments
        setIsCartUpdating={[MockFunction setIsCartUpdating]}
      />
    </div>
    <div>
      <div>
        <PriceSummary
          isUpdating={false}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`cartPage renders empty cart text (no adjustments, list or summary) if cart is empty 1`] = `
<div>
  <div>
    <h1>
      Cart
    </h1>
    <div>
      <StockStatusMessage />
    </div>
  </div>
  <div>
    <div>
      <h3>
        There are no items in your cart.
      </h3>
    </div>
    <div />
    <div>
      <div />
    </div>
  </div>
</div>
`;
